### PR TITLE
plan: support limit/group-by/order-by clauses in point get

### PR DESCRIPTION
**Pull request by jackysp**
### What problem does this PR solve? 
When the where clause can locate a unique row if the SQL has a limit/group-by/order-by it still can not use point get.
```
mysql> create table t (i int key, j int);
Query OK, 0 rows affected (0.12 sec)

mysql> explain select * from t where i = 1;
+-------------+-------+------+-------------------+
| id          | count | task | operator info     |
+-------------+-------+------+-------------------+
| Point_Get_1 | 1.00  | root | table:t, handle:1 |
+-------------+-------+------+-------------------+
1 row in set (0.00 sec)

mysql> explain select * from t where i = 1 limit 1;
+------------------------+-------+------+------------------------------------------------------+
| id                     | count | task | operator info                                        |
+------------------------+-------+------+------------------------------------------------------+
| Limit_8                | 1.00  | root | offset:0, count:1                                    |
| └─TableReader_13       | 1.00  | root | data:Limit_12                                        |
|   └─Limit_12           | 1.00  | cop  | offset:0, count:1                                    |
|     └─TableScan_11     | 1.00  | cop  | table:t, range:[1,1], keep order:false, stats:pseudo |
+------------------------+-------+------+------------------------------------------------------+
4 rows in set (0.00 sec)

mysql> explain select * from t where i = 1 order by i;
+--------------------+-------+------+-----------------------------------------------------+
| id                 | count | task | operator info                                       |
+--------------------+-------+------+-----------------------------------------------------+
| TableReader_12     | 1.00  | root | data:TableScan_11                                   |
| └─TableScan_11     | 1.00  | cop  | table:t, range:[1,1], keep order:true, stats:pseudo |
+--------------------+-------+------+-----------------------------------------------------+
2 rows in set (0.00 sec)

mysql> explain select * from t where i = 1 group by i;
+------------------------+-------+------+-----------------------------------------------------------------+
| id                     | count | task | operator info                                                   |
+------------------------+-------+------+-----------------------------------------------------------------+
| StreamAgg_17           | 1.00  | root | group by:col_2, funcs:firstrow(col_0), firstrow(col_1)          |
| └─TableReader_18       | 1.00  | root | data:StreamAgg_9                                                |
|   └─StreamAgg_9        | 1.00  | cop  | group by:test.t.i, funcs:firstrow(test.t.i), firstrow(test.t.j) |
|     └─TableScan_16     | 1.00  | cop  | table:t, range:[1,1], keep order:true, stats:pseudo             |
+------------------------+-------+------+-----------------------------------------------------------------+
4 rows in set (0.00 sec)
```

### What is changed and how it works?
Pass these clauses when try to use point get.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Side effects

 - Increased code complexity

PTAL @coocood @zz-jason

### CODE REVIEW

File: cmd/explaintest/t/explain_easy_stats.test [hunk 1]
```diff
@@ -48,5 +48,8 @@ explain select 1 in (select c2 from t2) from t1;
 explain format="dot" select 1 in (select c2 from t2) from t1;
 
 explain select * from index_prune WHERE a = 1010010404050976781 AND b = 26467085526790 LIMIT 1;
+explain select * from index_prune WHERE a = 1010010404050976781 AND b = 26467085526790 ORDER BY a;
+explain select * from index_prune WHERE a = 1010010404050976781 AND b = 26467085526790 GROUP BY b;
+explain select * from index_prune WHERE a = 1010010404050976781 AND b = 26467085526790 GROUP BY b ORDER BY a limit 1;
```
Comment (lamxTyler) [inline cmd/explaintest/t/explain_easy_stats.test]: What if it is limit 0?

Comment (jackysp) [reply cmd/explaintest/t/explain_easy_stats.test]: emm... I'll fix it.

File: plan/point_get_plan.go [hunk 2]
```diff
@@ -144,9 +145,15 @@ func tryFastPlan(ctx sessionctx.Context, node ast.Node) Plan {
...
 // 4. The condition is an access path that the range is a unique key.
 func tryPointGetPlan(ctx sessionctx.Context, selStmt *ast.SelectStmt) *PointGetPlan {
-	if selStmt.GroupBy != nil || selStmt.Having != nil || selStmt.OrderBy != nil || selStmt.Limit != nil ||
-		selStmt.LockTp != ast.SelectLockNone {
+	if selStmt.Having != nil || selStmt.LockTp != ast.SelectLockNone {
 		return nil
+	} else if selStmt.Limit != nil {
+		sc := ctx.GetSessionVars().StmtCtx
+		count, offset, err := extractLimitCountOffset(sc, selStmt.Limit)
+		logrus.Infof("YUSP count %d, offset %d", count, offset)
```
Comment (lamxTyler) [inline plan/point_get_plan.go]: Remove the log.

File: plan/point_get_plan.go [hunk 3]
```diff
@@ -144,9 +145,15 @@ func tryFastPlan(ctx sessionctx.Context, node ast.Node) Plan {
 // 3. All the columns must be public and generated.
 // 4. The condition is an access path that the range is a unique key.
 func tryPointGetPlan(ctx sessionctx.Context, selStmt *ast.SelectStmt) *PointGetPlan {
```
Comment (lamxTyler) [inline plan/point_get_plan.go]: The comment also needs to be updated.

**Comment by jackysp** [comment]
PTAL @lamxTyler

File: plan/point_get_plan.go [hunk 4]
```diff
@@ -139,14 +139,19 @@ func tryFastPlan(ctx sessionctx.Context, node ast.Node) Plan {
 // tryPointGetPlan determine if the SelectStmt can use a PointGetPlan.
 // Returns nil if not applicable.
 // To use the PointGetPlan the following rules must be satisfied:
-// 1. No group-by, having, order by, limit clause.
+// 1. No group-by, having, order by clause. For limit clause, the count should at least 1 and the offset is 0.
```
Comment (lamxTyler) [inline plan/point_get_plan.go]: Actually, we can have group-by and order-by clause.

Comment (jackysp) [reply plan/point_get_plan.go]: ah.. yes.

**Pull request merged**